plugins/ngx_jxl/README.md

```diff
@@ -0,0 +1,6 @@
+# NGINX JPEG XL modules
+This folder provides two modules for serving losslessly recompressed JPEG
+files in a transparent way.
+
+The `static` module serves `x.jpg.jxl` files instead of `x.jpg` files if they
```

Comment:
Could you expand on this documentation? what's the status of these plugins? compile instructions?
(how were these tested should be in the commit message).

For `static`, do we need to keep both the `x.jpg.jxl` and the `x.jpg`?

For the `filter`, I wonder why the choice of storing the .jpg in the server and compress to jxl on the fly instead of doing the opposite: storing the .jxl on the server and decompress to .jpg on the fly. The latter would save space on the server side and provide bandwidth savings (at the cost of CPU) but the former just costs more CPU for the server for only the bandwidth reduction so there's less incentive to use it.

Would it make sense to have a single plugin? It seems that having a single plugin would make it easier for people to use:
* if you request "foo.jpg" but accept image/jxl we can serve the cached foo.jpg.jxl; or transcode the foo.jpg to jxl (if for example an option in the plugin config enables that).
* if you request "foo.jxl" but don't accept image/jxl; we can serve the decoded jxl-to-jpg file as a image/jpg.

For reviewing it will be easier if we start with a smaller commit with just one module.

Reply:
I expanded the README.

The idea of the filter module is that you can enable it without thinking about it on existing webservers. Moreover, likely in a world where jxl support is not common it is preferable to spend more CPU for the JXL case (the CPU expenditure is remarkably similar either way).

As for a single plugin - I agree, but I don't know if that's at all possible. I assume that ngx_brotli has two plugins for a reason :)

For reviewing: the two plugins are entirely separate, so feel free to review only one of the plugins to begin with.